### DISCUSSION TITLE
Multiple network calls from same VM

### DISCUSSION THREAD

**Discussion by rubenquadros** [orbit-mvi/orbit-mvi#68] [Q&A] [answered]
Hi @Rosomack , 

Firstly, thanks for the awesome library!

I just started using this library and maybe this is a trivial question. 
`ContainerHost` requires `State` and `SideEffect`.
Now let's consider that in my UI/Screen I have to load an initial data and then based on user action I have to make another API call before doing further actions. 
But `ContainerHost` takes only one data class (like in example) so how do I pass on the second state data to UI?

I can definitely have extra params in my `State` data class like below:

`data class HomeScreenState(val count: Int?, val result: String?)`

Where `count` is response for first api and `result` is response for second api.

But is this the expected/right way?
Maybe my understanding is wrong. If so apologies in advance!

Eagerly waiting for your reply. Thank you!

**Comment by Rosomack** [comment, marked as answer]
@rubenquadros There are a few ways to do this, your suggestion is one of them. However, it can be cumbersome if there's a lot of nullable fields flying around.

You're free to use whatever you want for your state (as long as it's comparable) and one option is to define not one, but several states through using a sealed class and use `HomeScreenState` as the declared state for your `ContainerHost`  e.g.

```kotlin
sealed class HomeScreenState {
	object Loading: HomeScreenState()
	object Error: HomeScreenState()
	data class FirstDataState(val count): HomeScreenState()
	data class SecondDataState(val result): HomeScreenState()
}
```

Really it's up to you here - we want to do a deep dive into this topic at some point so that we have some more specific guidance.

**Reply by mochadwi** [reply]
Any suggestions or alternative when using `sealed class` inside `reduce` DSL?

```kotlin
reduce {
   state.copy // this is not possible when using `sealed class`
} 
```
